src/persist/src/persist.proto

```diff
@@ -70,3 +69,20 @@ message ProtoU64Description {
     ProtoU64Antichain upper = 2;
     ProtoU64Antichain since = 3;
 }
+
+message ProtoBatchInline {
+    ProtoBatchFormat format = 1;
+    oneof batch_type {
+        // To save bytes, only the seqno_lower and seqno_upper fields are set on
+        // this.
+        ProtoUnsealedBatchMeta unsealed = 2;
+        // To save bytes, only the desc field is set on this.
+        ProtoTraceBatchMeta trace = 3;
+    }
+}
+
+enum ProtoBatchFormat {
+    Unknown = 0;
+    ArrowKVTD = 1;
+    ParquetKVTD = 2;
```

Comment:
note to self: somewhere we should put a justification for the current structure. parquet would also allow something like `[(k, [(v, [(t, d)])])]` (whereas what's in the pr corresponds to something like `[(k, v, t, d)]`). the former would totally eliminate any duplication in keys and values and is similar in spirit to that OrderedLayers in differential's OrdKey/OrdVal batches. these each have tradeoffs that are worth examining at some point, but I don't think that point needs to be pre-1.0 (mostly because parquet per-column compression probably gets a lot of that back without the hassle). this format enum lets us make them in a backward compatible way

Reply:
Done